File: lab1web/NodejsConsoleApp2/Models/Tour.js
class Tour {
    //constructor(id = -1, name = null, country = null, price = null, maxTouristsCount = null, startDate = null) {
    //    this.id = id;
    //    this.name = name;
    //    this.country = country;
    //    this.price = price;
    //    this.maxTouristsCount = maxTouristsCount;
    //    this.startDate = startDate;
    //}
    constructor(item = null) {
        if (item != null) {
            this.id = item.id;
            this.name = item.name;
            this.country = item.country;
            this.price = item.price;
            this.maxTouristsCount = item.maxTouristsCount;
            this.startDate = item.startDate;
        }
        else {
            this.id = -1;
            this.name = null;
            this.country = null;
            this.price = null;
            this.maxTouristsCount = null;
            this.startDate = null;
        }
    }
}

module.exports = Tour;

File: weblab4/DAL.Abstractions/Models/Tour.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace DAL.Abstractions.Models
{
    public class Tour : IModel
    {
        public int TourId { get; set; }
        public string Name { get; set; }
        public string Country { get; set; }
        public decimal Price { get; set; }
        public int MaxTouristsCount { get; set; }
        public DateTime StartDate { get; set; }

        public virtual IEnumerable<Ticket> Tickets { get; set; }
        [NotMapped]
        public int KeyId => TourId;
    }
}
